tb-dev-challenge/tb-dev-challenge/ItemAddEditViewController.swift

```diff
@@ -0,0 +1,182 @@
+//
+//  ItemAddEditViewController.swift
+//  tb-dev-challenge
+//
+//  Created by Bhavin Kapadia on 2022-03-14.
+//
+import CoreData
+import UIKit
+
+class ItemAddEditViewController: UIViewController, UINavigationControllerDelegate {
+
+    @IBOutlet var menuItemNameTextField: UITextField!
+    @IBOutlet var menuItemPriceTextField: UITextField!
+    @IBOutlet var itemPriceValidationLabel: UILabel!
+    @IBOutlet weak var itemImagePreview: UIImageView!
+    @IBOutlet weak var deleteItemButton: UIButton!
+
+    var menuGroup: MenuGroups?
+    var imageSelected: String? = ""
+    var itemSelected: MenuItems?
+    let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        deleteItemButton.isHidden = true
+
+        if itemSelected != nil {
+            deleteItemButton.isHidden = false
+            deleteItemButton.addTarget(self, action: #selector(deleteItem), for: .touchUpInside)
+            menuItemNameTextField.text = itemSelected?.itemName
+            menuItemPriceTextField.text = "\(itemSelected?.itemPrice ?? 0.00)"
+            itemImagePreview.image = itemSelected?.itemImage?.toImage()
+        }
+
+        self.navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .save, target: self, action: #selector(createNewItemForGroup))
+        // MARK: Upload Button
+        let imageItemUploadButton = UIButton.init(type: .system)
+
+        imageItemUploadButton.frame = CGRect(x: 200,
+                                             y: 400,
+                                             width: 200,
+                                             height: 60)
+        imageItemUploadButton.setTitle("Upload Item Image",
+                                       for: .normal)
+
+        imageItemUploadButton.setTitleColor(.systemBlue,
+                                            for: .normal)
+        imageItemUploadButton.addTarget(self,
+                                        action: #selector(imageItemUploadAction),
+                                        for: .touchUpInside)
+        self.view.addSubview(imageItemUploadButton)
+
+        menuItemPriceTextField.delegate = self
+        setupView()
+
+    }
+
+    @objc func deleteItem() {
+        let menuItemToDelete = self.itemSelected
+        self.context.delete(menuItemToDelete!)
```

Comment:
probably also need to set `itemSelected` to nil after deletion

Reply:
fixed